### PR TITLE
ft: add cloudserver support for advanced options

### PR DESCRIPTION
**Pull request by alexanderchan-scality**

### CODE REVIEW

File: lib/management/configuration.js [hunk 1]
```diff
@@ -179,6 +179,8 @@ function patchConfiguration(newConf, log, cb) {
                         l.locationType });
                     return;
                 }
+                location.sizeLimitGB = l.sizeLimitGB || false;
```
Comment (rachedbenmustapha) [inline lib/management/configuration.js]: I'd expect `0 GB` to mean 'no limit' instead of `false GB` :)

Comment (jonathan-gramain) [inline lib/management/configuration.js]: I suggest leaving as `undefined` instead of `false` to avoid mixing data types (considered bad practice).

**Review by rachedbenmustapha — CHANGES_REQUESTED** [review]

File: lib/management/configuration.js (hunk 1 above)
Comment (alexanderchan-scality) [reply lib/management/configuration.js]: will fix

Comment (alexanderchan-scality) [reply lib/management/configuration.js]: will fix to use undefined for 0 value case

File: lib/management/configuration.js [hunk 2]
```diff
@@ -179,6 +179,8 @@ function patchConfiguration(newConf, log, cb) {
                         l.locationType });
                     return;
                 }
+                location.sizeLimitGB = l.sizeLimitGB || undefined;
```
Comment (rachedbenmustapha) [inline lib/management/configuration.js]: To ease debugging then, let's use `null` which conveys that it was manually assigned but chosen as missing-value, whereas `undefined` usually means that it was never manually defined.

**Review by rachedbenmustapha — CHANGES_REQUESTED** [review]

**Review by rachedbenmustapha — APPROVED** [review]

**Review by jonathan-gramain — APPROVED** [review]

**Pull request merged**